docs/manual/fread_examples.rst

```diff
@@ -0,0 +1,490 @@
+
+    data = """Name|Salary|Position
+             James|256,000|evangelist
+            Ragnar|1,000,000|conqueror
+              Loki|250360|trickster"""
+
+    fread(data)
+
+        Name	Salary	Position
+    0	James	256000	evangelist
+    1	Ragnar	1000000	conqueror
+    2	Loki	250360	trickster
+
+Specify the Delimiter
+---------------------
+
+You can specify the delimiter via the `sep` parameter.
+Note that the  separator must be a single character string; non-ASCII characters are not allowed as the separator, as well as any characters in ``["'`0-9a-zA-Z]``::
+
+    data = """
+           1:2:3:4
+           5:6:7:8
+           9:10:11:12
+           """
+
+    fread(data, sep=":")
+
+    	C0	C1	C2	C3
+    0	1	2	3	4
+    1	5	6	7	8
+    2	9	10	11	12
+
+Dealing with Null Values and Blank Rows
+---------------------------------------
+
+You can pass a list of values to be treated as null, via the `na_strings` parameter::
+
+    data = """
+           ID|Charges|Payment_Method
+           634-VHG|28|Cheque
+           365-DQC|33.5|Credit card
+           264-PPR|631|--
+           845-AJO|42.3|
+           789-KPO|56.9|Bank Transfer
+           """
+
+    fread(data, na_strings=['--', ''])
+
+        ID	    Charges  Payment_Method
+    0	634-VHG	    28	     Cheque
+    1	365-DQC	    33.5     Credit card
+    2	264-PPR	    631	     NA
+    3	845-AJO	    42.3     NA
+    4	789-KPO	    56.9     Bank Transfer
+
+
+For rows with null values, set `fill=True`::
+
+    data = ('a,b,c,d\n'
+            '1,2,3,4\n'
+            '5,6,7,8\n'
+            '9,10,11')
+
+    fread(data, fill=True)
+
+    	a	b	c	d
+    0	1	2	3	4
+    1	5	6	7	8
+    2	9	10	11	NA
+
+You can skip empty lines::
+
+    data = ('a,b,c,d\n'
+            '\n'
+            '1,2,3,4\n'
+            '5,6,7,8\n'
+            '\n'
+            '9,10,11,12')
+
+    fread(data, skip_blank_lines=True)
+
+        a	b	c	d
+    0	1	2	3	4
+    1	5	6	7	8
+    2	9	10	11	12
+
+Dealing with Column Names
+-------------------------
+
+If the data has no headers, `fread` will assign default column names::
+
+    data = ('1,2\n'
+            '3,4\n')
+
+    fread(data)
+
+        C0	C1
+    0	1	2
+    1	3	4
+
+You can pass in column names via the `columns` parameter::
+
+    fread(data, columns=['A','B'])
+
+        A	B
+    0	1	2
+    1	3	4
+
+You can change column names with the `columns` parameter::
+
+    data = ('a,b,c,d\n'
+            '1,2,3,4\n'
+            '5,6,7,8\n'
+            '9,10,11,12')
+
+    fread(data, columns=["A","B","C","D"])
+
+        A	B	C	D
+    0	1	2	3	4
+    1	5	6	7	8
+    2	9	10	11	12
+
+You can change `some` of the column names via a `dictionary`::
+
+    fread(data, columns={"a":"A", "b":"B"})
+
+        A	B	c	d
+    0	1	2	3	4
+    1	5	6	7	8
+    2	9	10	11	12
+
+By deafult, `fread` takes the first line in the data as the header. If, however, you do not want the first line as the header, set the `header` parameter to False::
```

Comment:
Fread will use heuristics to determine whether the first row is data or not (for example see your first example in this section). It may occasionally guess incorrectly, in which case `header=` parameter will be useful.